app/js/components/cohorts/cohortModal.js

```diff
@@ -68,13 +69,22 @@ class ActionsComponent extends Component {
                             name :'', 
                             description: ''
                     }));
+                    
                     $('#myCohort').modal('hide');          
                 });
         } else {
             this.setState({ error: "all fields are required" });
         }
     }
 
+    onSave(event) {
+        event.preventDefault();
+        const {name, description} = this.state;
+        if(name && description) {
```

Comment:
*If either of the fields has data, show an error but do no reset:* The error message says that both fields are required, so not resetting the input box gives the user an opportunity to fill in the missing field instead of refilling both fields.
*If both fields have data:* Save it and reset, so that the fields are empty for the next save.